Application/Events/InternalNotificationEventData.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+
+namespace Application\Events;
+
+
+
+use Domain\Interfaces\Services\Notifications\NotifiableInterface;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Notifications\Messages\MailMessage;
+use Illuminate\Notifications\Notification;
+
+class InternalNotificationEventData extends Notification implements ShouldQueue
+{
+    use Queueable;
+
+    private NotifiableInterface $notificationDetail;
+
+    public function __construct(NotifiableInterface $notification)
+    {
+        $this->notificationDetail = $notification;
+    }
+
+
+    public function via($notifiable)
+    {
+        return ['mail', 'database'];
```

Comment:
devolver un value object con prioridades no con las funciones concretas. depender de abstracciones nunca de implementaciones

Reply:
eso debería ser más o menos:
```suggestion
        return ['high' => 'mail', 'middle' => 'database', 'low' => 'database' ];
```